libs/core-ui/src/lib/Interfaces/IErrorAnalysisData.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+export interface IErrorAnalysisData {
+  tree?: IErrorAnalysisTreeNode[];
+  tree_features?: string[];
+  matrix?: IErrorAnalysisMatrix;
+  matrix_features?: string[];
+  maxDepth: number;
+  numLeaves: number;
+}
+// Represents the data retrieved from the backend
+export interface IErrorAnalysisTreeNode {
+  arg: number | number[] | undefined;
+  condition: string;
+  error: number;
+  id: number;
+  isErrorMetric: boolean;
+  method: string;
+  metricName: string;
+  metricValue: number;
+  nodeIndex: number;
+  nodeName: string;
+  parentId: number | undefined;
+  parentNodeName: string;
+  pathFromRoot: string;
+  size: number;
+  sourceRowKeyHash: string;
+  success: number;
+  badFeaturesRowCount: number;
+}
+
+export type IErrorAnalysisMatrixCategory =
+  | IErrorAnalysisMatrixClassificationCategory
```

Comment:
this shouldn't be classification vs regression, but error rate vs mean squared error, as more metrics will be added